src/com/amazon/ion/benchmark/DataConstructor.java

```diff
@@ -246,60 +255,76 @@ public static IonValue constructIonData(ReparsedType parsedTypeDefinition) {
-        for (Map.Entry<String, ReparsedType> entry : fieldMap.entrySet()) {
-            // Get the type definition for each field.
-            ReparsedType fieldTypeDefinition = entry.getValue();
-            // 'occurs' included in the field constraint determines the occurrences of the specified field.
-            int occurs = ReparsedType.getOccurs(fieldTypeDefinition.getConstraintStruct());
-            int occurTime = occurs == -1 ? random.nextInt(2) : occurs;
-            int i = 0;
-            while (i < occurTime) {
-                constructedIonStruct.add(entry.getKey(), constructIonData(fieldTypeDefinition));
-                i++;
+        if (element != null && fields != null) {
+            throw new IllegalStateException("Can only handle one of : " + VALID_STRUCT_CONSTRAINTS);
+        } else if (element != null) {
+            int length = container_length == null ? DEFAULT_CONTAINER_LENGTH : container_length.getRange().getRandomQuantifiableValueFromRange().intValue();
+            for (int i = 0; i < length; i++) {
+                constructedIonStruct.add(constructStringFromCodepointLength(random.nextInt(20)), constructIonData(element.getElement()));
+            }
+        } else {
+            Map<String, ReparsedType> fieldMap = fields.getFieldMap();
+            // Writing field value to IonStruct based on the relevant constraint.
+            for (Map.Entry<String, ReparsedType> entry : fieldMap.entrySet()) {
+                // Get the type definition for each field.
+                ReparsedType fieldTypeDefinition = entry.getValue();
+                // 'occurs' included in the field constraint determines the occurrences of the specified field.
+                int occurs = ReparsedType.getOccurs(fieldTypeDefinition.getConstraintStruct());
+                int occurTime = occurs == -1 ? random.nextInt(2) : occurs;
+                for (int i = 0; i < occurTime; i++) {
+                    constructedIonStruct.add(entry.getKey(), constructIonData(fieldTypeDefinition));
+                }
             }
         }
         return constructedIonStruct;
     }
 
     /**
-     * Constructing IonList which is aligned with the constraints provided in the constraintMap.
+     * Constructing IonSequence value which is aligned with the constraints provided in the constraintMap.
      * @param constraintMapClone collects the constraints from ISL file, the key represents the name of constraints,
      * and the value is constraint value in ReparsedConstraint format.
-     * @return the constructed IonList value.
+     * @param container represents one of the element from set (IonList | IonSexp).
+     * @return the constructed IonSequence value.
      */
-    private static IonList constructIonList(Map<String, ReparsedConstraint> constraintMapClone) {
+    private static IonValue constructSequenceTypeData(Map<String, ReparsedConstraint> constraintMapClone, IonSequence container) {
```

Comment:
Should this return `IonSequence`? Or, should this be a templated method that returns the same type as `container`?

Reply:
This method should return IonSequence for now.  It will be updated in the following commit.